presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java

```diff
@@ -95,6 +96,23 @@ private SymbolsExtractor() {}
         return ImmutableSet.copyOf(extractAllVariable(expression, types));
     }
 
+    public static Set<VariableReferenceExpression> extractUniqueVariable(RowExpression expression, TypeProvider types)
```

Comment:
These two functions are not necessary and actually improper to be called by optimizers. Check my comment below.